src/data/chats/chat.rs

```diff
@@ -343,4 +349,8 @@ impl Chat {
             .unwrap();
         self.messages.truncate(message_index);
     }
+
+    pub fn touch_last_selected_at(&mut self) {
+        self.last_selected_at = chrono::Utc::now();
+    }
```

Comment:
I think this naming is flawed. We are on the Chat struct, the concept of "touch" doesn't really exist at this level of abstraction, it also couples it too much with the caller. Furthermore, i feel like saving "selection" is a bit weird, why would a chat need to remember if it was selected or not? that's a caller/UI level concept. wouldn't it be better to save a last updated or something like that, and interpret a selection as a form of update from the outside? If anything i would keep the idea of a selected chat in the Chats struct and above, does this make sense?